Design.md

```diff
@@ -253,3 +280,23 @@ Get the platform usage of global (without query parameter), tenant (with tenant_
 ## S3 Operations Example Activity Diagram
 ![S3 Object operations](Design_Files/OSE-OSIS-Object-operations.png)
 
+## Cache Design
+
+1. Implement caches using the following principles:
+    1. Cache must have a maximum capacity.
+    1. Cache must have a Least Recently Used (LRU) eviction policy.
+    1. Each cache entry must have an expiration time (TTL). 
+        1. Cache `put` call must include `expirationTime` for the entry.
+        1. Use `ScheduledFuture` in Java for cache entry removal after TTL.
+    1. Caches must handle concurrency.
+        1. Use `Read locks` for `get` calls and `write locks` for `put` calls.
+    1. Caches can be configured using the following properties in `application.properties`:
+       1. `expirationTime` for invalidating cache entries
```

Comment:
Is `expirationTime` in seconds?
If yes, `durationSeconds` is a better variable. The same was done in vaultClient.
https://github.com/scality/vaultclient/pull/294/files#diff-6ed3f39879bec9944e1a6e9d0d3c1bcbc7209a7eab826e5c3819048e4841163dR136

Reply:
No this milliseconds

Reply:
Okay then it's not `expirationTime`. It's the duration.

Reply:
I am not sure if `duration` is a right term for cache. For cache usually we name it `ttl` or `expiration` right.

Reply:
`ttl` / `expiration` sound good.

But as this is in units of seconds, its not `expirationTime` unless its epoch time at which creds would expire.

`durationSeconds` was just a suggestion as we use the same in AWS / Vault

Reply:
No `ttl` is not time. It is usually in units like 3600000 ms for 1 hour

Reply:
done 569dba03652832856a3dc38329c6cbf4872fa1d4